### PR TITLE
Suggestions to #219

### PR DESCRIPTION
**Pull request by chapulina**
Some suggestions while reviewing the PR:

* Added more tests for `Plane::Intersect` (but I think we should add more once the plane's size is taken into consideration)
* Extended `Line3::Distance` to consider the line's limits (see `Vector3::DistToLine` for an existing function that works with infinite lines, also added more tests there)
* Made some updates to `Sphere::VolumeBelow` to handle vertical planes, and I also believe that the function was originally returning the volume above, see the added tests.
* Started adding tests for `Sphere::CenterOfVolumeBelow`
* Small style updates

I'm still reviewing the PR, I haven't looked into `Box` yet.

### CODE REVIEW

File: include/ignition/math/detail/Sphere.hh [hunk 1]
```diff
@@ -103,18 +103,24 @@ T Sphere<T>::VolumeBelow(const Planed &_plane) const
...
+  else if (dist > r)
   {
     // sphere is completely above plane
-    return 0;
+    return 0.0;
   }
 
-  auto h = r + dist;
+  // Vertical plane
+  if (_plane.Normal().Z() < 1e-6)
```
Comment (arjo129) [inline include/ignition/math/detail/Sphere.hh:118]: Why should this be the case? I consider "below" to be defined by the normal. It is possible that the sphere is rotated when surfacing.

**Review by arjo129 — COMMENTED** [review]
Thanks for the :bug: fixes! Most of the changes look good to me. However, I don't agree with the sphere and the vertical plane. I guess I need to be more articulate in my documentation but volume below should return the Volume "below" the plane where "up" refers to the normal of the line. The rationale for this is in future revisions of the buoyancy plugin, I'd like to be rotate the object/plane according to its pose.

File: include/ignition/math/detail/Sphere.hh (hunk 1 above)
Comment (chapulina) [reply include/ignition/math/detail/Sphere.hh:118]: Ah ok, I assumed the opposite, and even updated the docs :sweat_smile: 

https://github.com/ignitionrobotics/ign-math/blob/f398782fd650547d2d1be9693a9e7569bbe38e3a/include/ignition/math/Sphere.hh#L97

In that case, I think the implementation needs to be fixed to reflect that. See this test that I added, the normal doesn't seem to affect the result:

https://github.com/ignitionrobotics/ign-math/blob/f398782fd650547d2d1be9693a9e7569bbe38e3a/src/Sphere_TEST.cc#L141-L149

Comment (chapulina) [reply include/ignition/math/detail/Sphere.hh:118]: Reading that doc update now, I think it's badly worded :grimacing: I meant to say that the normal's direction is not taken into account.

Comment (arjo129) [reply include/ignition/math/detail/Sphere.hh:118]: Yep will need to fix that. I can merge these changes in and then work on a fix for it.

**Review by arjo129 — APPROVED** [review]

**Pull request merged**